app/repositories/application/PreviousYearCandidatesDetailsRepository.scala

```diff
@@ -0,0 +1,498 @@
+    "reportLinkURL," +
+  "Situational T-score,Situational Percentile,Situational Raw,Situational STEN," +
+  "PHASE_2 scheduleId,cubiksUserId,token,testUrl,invitiationDate,participantScheduleId,startedDateTime,completedDateTime,reportLinkURL,reportId," +
+    "e-Tray T-score,e-Tray Raw,interviewId,token,candidateId,customCandidateId,comment,Q1 Capability,Q1 Engagement,Q2 Capability,Q2 Engagement,Q3 Capability," +
+    "Q3 Engagement,Q4 Capability,Q4 Engagement,Q5 Capability,Q5 Engagement,Q6 Capability,Q6 Engagement,Q7 Capability," +
+    "Q7 Engagement,Q8 Capability,Q8 Engagement,Overall total," +
+    "IN_PROGRESS,SUBMITTED,PHASE1_TESTS_INVITED,PHASE1_TESTS_STARTED,PHASE1_TESTS_COMPLETED,PHASE1_TESTS_RESULTS_READY," +
+    "PHASE1_TESTS_RESULTS_RECEIVED,PHASE1_TESTS_PASSED,PHASE2_TESTS_INVITED,PHASE2_TESTS_FIRST_REMINDER," +
+    "PHASE2_TESTS_SECOND_REMINDER,PHASE2_TESTS_STARTED,PHASE2_TESTS_COMPLETED,PHASE2_TESTS_RESULTS_READY," +
+    "PHASE2_TESTS_RESULTS_RECEIVED,PHASE2_TESTS_PASSED,PHASE3_TESTS_INVITED,PHASE3_TESTS_FIRST_REMINDER," +
+    "PHASE3_TESTS_SECOND_REMINDER,PHASE3_TESTS_STARTED,PHASE3_TESTS_COMPLETED,PHASE3_TESTS_RESULTS_RECEIVED," +
+    "PHASE3_TESTS_PASSED,PHASE3_TESTS_SUCCESS_NOTIFIED,EXPORTED,PHASE 1 result,result,result,result,result,result,result,result,result,result,result,result,result,result," +
+    "result,result,result," +
+    "PHASE 2 result,result,result,result,result,result,result,result,result,result,result,result,result,result,result,result,result," +
+    "PHASE 3 result,result,result,result,result,result,result,result,result," +
+    "result,result,result,result,result,result,result,result"
+
+  val contactDetailsHeader = "Email,Address line1,Address line2,Address line3,Address line4,Postcode,Outside UK,Country,Phone"
+
+  val questionnaireDetailsHeader = "Gender Identity,Sexual Orientation,Ethnic Group,Live in UK between 14-18?,Home postcode at 14," +
+    "Name of school 14-16,Name of school 16-18,Eligible for free school meals?,University name,Category of degree," +
+    "Parent guardian completed Uni?,Parents job at 14,Employee?,Size," +
+    "Supervise employees,SE 1-5,Oxbridge,Russell Group,Hesa Code"
+
+  val mediaDetailsHeader = "How did you hear about us?"
+
+  def applicationDetailsStream(): Enumerator[CandidateDetailsReportItem]
+
+  def findContactDetails(): Future[CsvExtract[String]]
+
+  def findQuestionnaireDetails(): Future[CsvExtract[String]]
+}
+
+class PreviousYearCandidatesDetailsMongoRepository(implicit mongo: () => DB) extends PreviousYearCandidatesDetailsRepository with CommonBSONDocuments {
+  import config.MicroserviceAppConfig._
+
+  val applicationDetailsCollection = mongo().collection[JSONCollection](CollectionNames.APPLICATION)
+
+  val contactDetailsCollection = mongo().collection[JSONCollection](CollectionNames.CONTACT_DETAILS)
+
+  val questionnaireCollection = mongo().collection[JSONCollection](CollectionNames.QUESTIONNAIRE)
+
+  private def optYes = Some("Yes")
```

Comment:
Would an implicit remove the need to check booleans and return these values everywhere?